torch/_inductor/codegen/cpp.py

```diff
@@ -355,9 +359,13 @@ def store(self, name, index, value, mode=None):
 
     def reduction(self, name, dtype, src_dtype, reduction_type, index, value):
         argmax_or_argmin = reduction_type in {"argmax", "argmin"}
+        need_accmulate = reduction_type == "sum"
         tmpvar = self.cse.generate(
             self.loads, f"reduction {name} {cexpr(index)}", write=False
         )
+        if need_accmulate:
+            max_threads = self.num_threads if self.num_threads is not None else torch.get_num_threads()
+            tmpvar_to_accumulate = self.cse.var_to_accumulate(tmpvar, max_threads)
```

Comment:
Can this `var_to_acc` be kept inside cpp codegen? I think it is specific to cpp?